### PR TITLE
Support running as a Windows service.

### PR DESCRIPTION
**Pull request by hvvvvvvv**
When the program is started by the Service Control Manager (SCM), it needs to call `StartServiceCtrlDispatcher` (which corresponds to `service_dispatcher::start` in the windows-service crate) to inform the SCM of the service's entry function. The SCM then calls the service entry function passed to `StartServiceCtrlDispatcher`. The process calling `StartServiceCtrlDispatcher` will block until the service's status is set to Stopped. If the current program is not run through the SCM, `StartServiceCtrlDispatcher` will return the error `ERROR_FAILED_SERVICE_CONTROLLER_CONNECT`, and the program will run according to its original mechanism. For more details about SCM, please refer to [Microsoft's documentation](https://msdn.microsoft.com/en-us/library/windows/desktop/ee126211(v=vs.85).aspx).

### CODE REVIEW

File: easytier/src/easytier-core.rs [hunk 1]
```diff
@@ -638,13 +640,109 @@ pub fn handle_event(mut events: EventBusSubscriber) -> tokio::task::JoinHandle<(
...
+}
+
 #[tokio::main]
 async fn main() {
     setup_panic_handler();
 
     let locale = sys_locale::get_locale().unwrap_or_else(|| String::from("en-US"));
     rust_i18n::set_locale(&locale);
 
+    #[cfg(target_os = "windows")]
```
Comment (KKRainbow) [inline easytier/src/easytier-core.rs]: what will happen if not started as a service (e.g. from cmd / powershell)

Comment (KKRainbow) [reply easytier/src/easytier-core.rs]: i think we need add a cli argument to indicate if we launched by windows service mgr

**Comment by KKRainbow** [comment]
also please fill the PR description, explaining why we need this patch and what will happen without this patch when running as windows service.

File: easytier/src/easytier-core.rs (hunk 1 above)
Comment (hvvvvvvv) [reply easytier/src/easytier-core.rs]: > i think we need add a cli argument to indicate if we launched by windows service mgr

Currently, I have only added support for running through the Service Control Manager (SCM). easytier-Core can be installed as a Windows service using sc.exe. In the future, I will add functionality for installing and uninstalling the service through the program's own CLI, as well as other service management features.

File: easytier/Cargo.toml [hunk 2]
```diff
@@ -189,6 +189,7 @@ windows-sys = { version = "0.52", features = [
 ] }
 encoding = "0.2"
 winreg = "0.52"
+windows-service = "0.7.0"
```
Comment (KKRainbow) [inline easytier/Cargo.toml:192]: make this dep windows-only please

File: easytier/src/easytier-core.rs [hunk 3]
```diff
@@ -2,8 +2,7 @@
 extern crate rust_i18n;
 
 use std::{
-    net::{Ipv4Addr, SocketAddr},
-    path::PathBuf,
+    ffi::OsString, net::{Ipv4Addr, SocketAddr}, path::PathBuf
```
Comment (KKRainbow) [inline easytier/src/easytier-core.rs]: wrap it into #[cfg] block

File: easytier/src/easytier-core.rs (hunk 1 above)
Comment (HKMV) [reply easytier/src/easytier-core.rs]: 可以参考我写的项目https://github.com/hkmv/nal

**Pull request merged**